utilities/tools/Test-ModuleLocally.ps1

```diff
@@ -0,0 +1,181 @@
+                        moduleFolderPaths = (Join-Path $PSScriptRoot '..\..\arm' $ModuleName)
+                    }
+                }
+                Filter     = @{
+                    #ExcludeTag = 'ApiCheck'
+                    #Tag = 'ApiCheck'
+                }
+                TestResult = @{
+                    TestSuiteName = 'Global Module Tests'
+                    Enabled       = $false
+                }
+                Output     = @{
+                    Verbosity = 'Detailed'
+                }
+            }
+        }
+        # Deploy Module
+        if (($ValidateOnly -or $DeployTest) -and $ValidateOrDeployParameters) {
+            Write-Verbose '---- Deploying Module ----'
+            # Find Test Parameter Files
+            $ModuleParameterFiles = Get-ChildItem -Path (Join-Path $PSScriptRoot '..\..\arm' $ModuleName '.parameters') -Recurse
+            # Load Tokens Converter Script
+            . (Join-Path $PSScriptRoot '..\..' '.github\actions\sharedScripts\Convert-TokensInFileList.ps1')
+            # Replace Tokens with Values For Local Testing
+            $TokensReplaceWithObject = @(
+                @{ Replace = '<<tenantId>>'; With = "$($ValidateOrDeployParameters.TenantId)" }
+                @{ Replace = '<<subscriptionId>>'; With = "$($ValidateOrDeployParameters.SubscriptionId)" }
+                @{ Replace = '<<managementGroupId>>'; With = "$($ValidateOrDeployParameters.ManagementGroupId)" }
+                @{ Replace = '<<principalId1>>'; With = "$($ValidateOrDeployParameters.PrincipalId)" }
+            )
+            $ModuleParameterFiles | ForEach-Object { Convert-TokensInFileList -Paths $PSitem.FullName -TokensReplaceWith $TokensReplaceWithObject }
+
+            # Build Modules Validation and Deployment Inputs
+            $functionInput = @{
+                templateFilePath  = (Join-Path $PSScriptRoot '..\..\arm' $ModuleName 'deploy.bicep')
+                parameterFilePath = (Join-Path $PSScriptRoot '..\..\arm' $ModuleName '.parameters\parameters.json')
+                location          = "$($ValidateOrDeployParameters.Location)"
+                resourceGroupName = "$($ValidateOrDeployParameters.ResourceGroupName)"
+                subscriptionId    = "$($ValidateOrDeployParameters.SubscriptionId)"
+                managementGroupId = "$($ValidateOrDeployParameters.ManagementGroupId)"
+            }
+
+            try {
+                # Validate Template
+                if ($ValidateOnly) {
+                    # Load Modules Deployment Script
+                    . (Join-Path $PSScriptRoot '..\..' '.github\actions\templates\validateModuleDeploy\scripts\Test-TemplateWithParameterFile.ps1')
+                    # Invoke Validation
+                    Test-TemplateWithParameterFile @functionInput -Verbose
+                }
+
+                # Deploy Template
+                if ($DeployTest -and !$ValidateOnly) {
```

Comment:
Any reason for the if-else?  If somebody deliberatly uses both switches I guess it would be fine to also run both blocks in squence

Reply:
That's fine. I will rename the parameters to make it more clear then